src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalLocaleOpt.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n.messages.api.opt.local;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vmware.vipclient.i18n.VIPCfg;
+import com.vmware.vipclient.i18n.messages.api.opt.LocaleOpt;
+
+public class LocalLocaleOpt implements LocaleOpt{
+
+    private Logger logger = LoggerFactory.getLogger(LocalLocaleOpt.class);
+    private static final String BUNDLE_PREFIX = "messages_";
+    
+    @Override
+    public Map<String, String> getLanguages(String displayLanguage) {
+   
+    	Map<String, String> supportedLocales = new HashMap<String, String>();
+    	Locale inLocale = Locale.forLanguageTag(displayLanguage); 
+    	String offlineResourcesBaseUrl = VIPCfg.getInstance().getOfflineResourcesBaseUrl();
+		try {
+			Path path = Paths.get(Thread.currentThread().getContextClassLoader().
+					getResource(Paths.get(offlineResourcesBaseUrl).toString()).toURI());
+			
+			try (Stream<Path> listOfFiles = Files.walk(path).filter(Files::isRegularFile)) {
+				listOfFiles.map(file -> {
+					String fileName = file.getFileName().toString();
+					return fileName.substring(BUNDLE_PREFIX.length(), fileName.indexOf("."));
```

Comment:
```suggestion
					return fileName.substring(BUNDLE_PREFIX.length(), fileName.indexOf('.'));
```
Put single-quotes around '.' to use the faster "indexOf(char)" method.